src/Symfony/Component/Translation/Formatter/IntlMessageFormatter.php

```diff
@@ -12,27 +12,50 @@
 namespace Symfony\Component\Translation\Formatter;
 
 use Symfony\Component\Translation\Exception\InvalidArgumentException;
+use Symfony\Component\Translation\Exception\LogicException;
 
 /**
  * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
  * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
  */
-class IntlMessageFormatter implements MessageFormatterInterface
+class IntlMessageFormatter implements MessageFormatterInterface, IntlFormatterInterface
 {
+    private $hasMessageFormatter;
+    private $cache = array();
+
     /**
      * {@inheritdoc}
      */
     public function format($message, $locale, array $parameters = array())
     {
-        try {
-            $formatter = new \MessageFormatter($locale, $message);
-        } catch (\Throwable $e) {
-            throw new InvalidArgumentException(sprintf('Invalid message format (%s, error #%d).', intl_get_error_message(), intl_get_error_code()), 0, $e);
+        return $this->formatIntl($message, $locale, $parameters);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function formatIntl(string $message, string $locale, array $parameters = array()): string
+    {
+        if (!$formatter = $this->cache[$locale][$message] ?? null) {
+            if (!($this->hasMessageFormatter ?? $this->hasMessageFormatter = class_exists(\MessageFormatter::class))) {
+                throw new LogicException('Cannot parse message translation: please install the "intl" PHP extension or the "symfony/polyfill-intl-messageformatter" package.');
+            }
+            try {
+                $this->cache[$locale][$message] = $formatter = new \MessageFormatter($locale, $message);
+            } catch (\IntlException $e) {
+                throw new InvalidArgumentException(sprintf('Invalid message format (error #%d): %s.', intl_get_error_code(), intl_get_error_message()), 0, $e);
+            }
+        }
+
+        foreach ($parameters as $key => $value) {
+            if ('%' === ($key[0] ?? null)) {
+                unset($parameters[$key]);
+                $parameters[trim($key, '%')] = $value;
```

Comment:
should we also support the `{{ name }}` convention, which is used in the Validator component for instance ?

Reply:
good idead, updated